src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -356,69 +362,171 @@ private static unsafe void GetCursorPosition(out int left, out int top)
+                    // as easy it it sounds.  Prior to the CPR having been supplied to stdin, other
+                    // user input could have come in and be available to read first from stdin.  Plus,
+                    // that user input could include escape sequences, and those escape sequences could
+                    // have a prefix very similar to that of the CPR (e.g. other escape sequences start
+                    // with \ESC + '['.  It's also possible that some terminal implementations may not
+                    // write the CPR to stdin atomically, such that the CPR could have other user input
+                    // in the middle of it, and that user input could have escape sequences!  Handling
+                    // that last case is very challenging, and rare, so we don't try, but we do need to
+                    // handle the rest.  The min bar here is doing something reasonable, which may include
+                    // giving up and just returning default top and left values.
+
+                    // Consume from stdin until we find all of the key markers for the CPR:
+                    // \ESC, '[', ';', and 'R'.  For everything before the \ESC, it's definitely
+                    // not part of the CPR sequence, so we just immediately move any such bytes
+                    // over to the StdInReader's extra buffer.  From there until the end, we buffer
+                    // everything into readBytes for subsequent parsing.
+                    const byte Esc = 0x1B;
                     StdInReader r = StdInReader.Inner;
-                    byte b;
+                    int escPos, bracketPos, semiPos, rPos;
+                    if (!AppendToStdInReaderUntil(Esc, r, readBytes, ref readBytesPos, out escPos) ||
+                        !BufferUntil((byte)'[', r, ref readBytes, ref readBytesPos, out bracketPos) ||
+                        !BufferUntil((byte)';', r, ref readBytes, ref readBytesPos, out semiPos) ||
+                        !BufferUntil((byte)'R', r, ref readBytes, ref readBytesPos, out rPos))
+                    {
+                        // We were unable to read everything from stdin, e.g. a timeout ocurred.
+                        // Since we couldn't get the complete CPR, transfer any bytes we did read
+                        // back to the StdInReader's extra buffer, treating it all as user input,
+                        // and exit having not computed a valid cursor position.
+                        TransferBytes(readBytes.Slice(readBytesPos), r);
+                        return;
+                    }
+
+                    // At this point, readBytes starts with \ESC and ends with 'R'.
+                    Debug.Assert(readBytesPos > 0 && readBytesPos <= readBytes.Length);
+                    Debug.Assert(escPos == 0 && bracketPos > escPos && semiPos > bracketPos && rPos > semiPos);
+                    Debug.Assert(readBytes[escPos] == Esc);
```

Comment:
Just wondering, is it possible that any of the following four asserts are ever possible?